test/integration/targets/ansible-galaxy-collection/tasks/upgrade.yml

```
@@ -0,0 +1,166 @@
+##### Updating collections with a new version constraint
+
+# No deps
+
+- name: reset installation directory
+  file:
+    state: "{{ item }}"
+    path: "{{ galaxy_dir }}/ansible_collections"
+  loop:
+    - absent
+    - directory
+
+- name: install a collection
+  command: ansible-galaxy collection install namespace1.name1:0.0.1 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+  failed_when:
+    - '"namespace1.name1:0.0.1 was installed successfully" not in result.stdout_lines'
+
+- name: install a new version of the collection without --force
+  command: ansible-galaxy collection install namespace1.name1:>0.0.4,<=0.0.5 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: '"namespace1.name1:0.0.5 was installed successfully" in result.stdout_lines'
+
+- name: don't reinstall the collection in the requirement is compatible
+  command: ansible-galaxy collection install namespace1.name1:>=0.0.5 -s pulp_v3 {{ galaxy_verbosity }}
+  register: result
+
+- assert:
+    that: "\"Nothing to do. All requested collections are already installed.\" in result.stdout"
```

Pro tip: this is how you can avoid escaping quotes:
```suggestion
    that: >-
      "Nothing to do. All requested collections are already installed." in result.stdout
```

---

Yeah, that's a nice tip (you've mentioned it before). I consciously chose the one-liner.

---

You could also just wrap the entire line with single quotes but any of these methods are just fine.